src/lazy_buffer.rs

```diff
@@ -44,6 +40,18 @@ where
             }
         }
     }
+
+    pub fn prefill(&mut self, len: usize) {
+        if !self.done && len > self.buffer.len() {
+            for _ in 0..(len - self.buffer.len()) {
+                if let Some(elt) = self.it.next() {
+                    self.buffer.push(elt);
+                } else {
+                    self.done = true;
+                }
+            }
+        }
+    }
```

Comment:
Calling `.next()` in a loop performs very poorly for iterators like `Chain`. Could you avoid that by writing something like this, maybe?
```suggestion
    pub fn prefill(&mut self, len: usize) {
        let buffer_len = self.buffer.len();
        if !self.done && len > buffer_len {
            let delta = len - buffer_len;
            self.buffer.extend(self.it.by_ref().take(delta));
            self.done = self.buffer.len() == len;
        }
    }
```

Reply:
> Calling `.next()` in a loop performs very poorly for iterators like `Chain`. Could you avoid that by writing something like this, maybe?

Interesting to know. I had been looking for a way of doing this but didn't think to combine `by_ref()` with `take()`. Useful technique!

I'll use this except the final line should be: `self.done = self.buffer.len() < len;` I.e. set `self.done` if the buffer wasn't filled to length `len`.